src/cloudai/_core/base_installer.py

```diff
@@ -168,8 +209,13 @@ def uninstall(self, items: Iterable[Installable]) -> InstallStatusResult:
         """
         logging.debug(f"Going to uninstall {len(set(items))} uniq items (total {len(list(items))}).")
         logging.info(f"Going to uninstall {len(set(items))} items.")
+
+        max_workers = 1 if self.is_low_thread_environment else None
```

Comment:
Let's have `self.num_workers` property to reduce code duplication.
BTW, we can me logging line into `is_low_thread_environment` to avoid double printing.